Sources/Sentry/SentryUIViewControllerPerformanceTracker.m

```diff
@@ -18,12 +18,26 @@
 #    import <UIKit/UIKit.h>
 #    import <objc/runtime.h>
 
+// Instead of using associated objects, we use a global map to store the time to display tracker,
+// spanId, spans in execution, and layout subview spanId to avoid memory issues with associated
+// objects accessed from different threads.
+//
+// Using a NSMapTable allows weak references to the keys, which means we don't need to remove the
+// entries when the UIViewController is deallocated.
```

Comment:
`l`: I would put that comment directly above the NSMapTable properties, so they are colocated.